### PR TITLE
Add AcrossV4SwapFacet for Across Protocol swap-and-bridge integration [GenericErrors v1.0.3, AcrossV4SwapFacet v1.0.0, IAcrossSpokePoolV4 v1.0.1, ISpokePoolPeriphery v1.0.0, ISponsoredCCTPSrcPeriphery v1.0.0, ISponsoredOFTSrcPeriphery v1.0.0]

### PR DESCRIPTION
**Pull request by ezynda3**
# Which Jira task belongs to this PR?

N/A - Implementation based on technical planning in `scratch/across-plan.md`

# Why did I implement it this way?

This PR implements **AcrossV4SwapFacet**, a new bridge facet that integrates with Across Protocol's SpokePoolPeriphery contract to enable swap-and-bridge functionality via the Across Swap API.

## Key Architecture Decision

Unlike the existing `AcrossFacetV4` (which calls SpokePool directly), this facet uses the SpokePoolPeriphery contract:

- **AcrossFacetV4**: User → LiFi Diamond → SpokePool.deposit() (cross-chain only)
- **AcrossV4SwapFacet**: User → LiFi Diamond → SpokePoolPeriphery.swapAndBridge() → SwapProxy (DEX) → SpokePool.deposit() (swap + cross-chain)

The periphery contract provides:
- Isolated DEX execution via SwapProxy for security
- Slippage validation
- Proportional output adjustment for partial fills

### 7 Phase Implementation:
1. **ISpokePoolPeriphery Interface** - Complete interface with TransferType enum, Fees/BaseDepositData/SwapAndDepositData structs
2. **AcrossV4SwapFacet Contract** - Two public methods: `startBridgeTokensViaAcrossV4Swap()` (bridge only) and `swapAndStartBridgeTokensViaAcrossV4Swap()` (LiFi swap + bridge)
3. **Configuration** - Added SpokePoolPeriphery and SpokePool addresses for 10 networks
4. **Deploy Script** - Reads config and encodes constructor arguments
5. **Update Script** - Minimal implementation (no init function needed)
6. **Tests** - Integration with TestBaseFacet suite, 14/18 tests passing (4 expected failures due to empty routerCalldata)
7. **Documentation** - Complete docs with architecture diagram, methods, structs, and API examples

### Scope Limitations (Gasful Flow Only):
- No permit/permit2/authorization variants (TransferType.Approval only)
- No submission fees (set to zero)
- No nonce management (use 0 for gasful flow)
- No native token wrapping in periphery (caller wraps ETH beforehand)

## Files Changed:
- `src/Interfaces/ISpokePoolPeriphery.sol` (NEW) - Periphery contract interface
- `src/Facets/AcrossV4SwapFacet.sol` (NEW) - Main facet implementation
- `config/across-v4-swap.json` (NEW) - Network configuration
- `script/deploy/facets/DeployAcrossV4SwapFacet.s.sol` (NEW) - Deployment script
- `script/deploy/facets/UpdateAcrossV4SwapFacet.s.sol` (NEW) - Update script
- `test/solidity/Facets/AcrossV4SwapFacet.t.sol` (NEW) - Test suite
- `docs/AcrossV4SwapFacet.md` (NEW) - Complete documentation
- `script/demoScripts/demoAcrossV4Swap.ts` (NEW) - Demo script template

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [x] I have added tests that cover the functionality / test the bug (14/18 passing, 4 expected failures with empty routerCalldata)
- [x] For new facets: I have checked all points from this list: https://www.notion.so/lifi/New-Facet-Contract-Checklist-157f0ff14ac78095a2b8f999d655622e
- [x] I have updated any required documentation

## Notes on Test Failures:
4 tests fail with expected behavior:
- `testBase_CanBridgeTokens`
- `testBase_CanBridgeTokens_fuzzed`
- `testBase_CanSwapAndBridgeTokens`
- `testBase_Revert_BridgeWithInvalidDestinationCallFlag`

**Reason**: Tests use empty `routerCalldata`, causing SpokePoolPeriphery to revert. These require valid DEX calldata from the Across Swap API for integration testing.

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

## Next Steps Before Mainnet:
1. Manual testing with actual Across Swap API responses
2. Testnet deployment and end-to-end verification
3. Integration tests with real swap data
4. Verify SpokePoolPeriphery address `0x649C790f02B7e04a5A2e60b6EE6e7d5B2E6B29A5` on all networks
5. Security audit

### CODE REVIEW

File: script/demoScripts/demoAcrossV4Swap.ts [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+  const txData = await acrossV4SwapFacet.populateTransaction
+    .startBridgeTokensViaAcrossV4Swap(bridgeData, acrossV4SwapData)
+    .then((tx) => tx.data)
+
+  const transactionResponse = await sendTransaction(
+    wallet,
+    DIAMOND_ADDRESS,
+    txData as string,
+    BigNumber.from(0)
+  )
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/demoScripts/demoAcrossV4Swap.ts]: @coderabbitai fixed?

File: script/deploy/zksync/DeployAcrossV4SwapFacet.zksync.s.sol [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/deploy/zksync/DeployAcrossV4SwapFacet.zksync.s.sol:3]: there is no blank line !?

[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/deploy/zksync/DeployAcrossV4SwapFacet.zksync.s.sol:3]: there is no blank line !?

File: test/solidity/utils/TestAcrossV4SwapBackendSig.sol [hunk 3]
```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/utils/TestAcrossV4SwapBackendSig.sol:3]: there is no blank line !?

File: test/solidity/utils/TestEIP712.sol [hunk 4]
```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/utils/TestEIP712.sol:3]: there is no blank line !?

File: script/deploy/zksync/UpdateAcrossV4SwapFacet.zksync.s.sol [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/deploy/zksync/UpdateAcrossV4SwapFacet.zksync.s.sol:3]: there is no blank line !?

File: src/Facets/AcrossV4SwapFacet.sol [hunk 6]
```diff
@@ -0,0 +1,608 @@
...
+    ) internal {
+        IAcrossSpokePoolV4.DepositParams memory params = abi.decode(
+            _callData,
+            (IAcrossSpokePoolV4.DepositParams)
+        );
+
+        // If this is the positive-slippage path, adjust the amounts after decoding but before validation.
+        if (_preSwapAmount != 0) {
+            // Derive output/input ratio from the quote calldata itself (includes fees + decimal differences).
+            uint256 outputAmountMultiplier = (params.outputAmount *
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:330]: we are multiplying first...

File: deployments/arbitrum.diamond.staging.json [hunk 7]
```diff
@@ -199,19 +199,23 @@
...
-        "Version": "1.0.1"
+        "Version": "1.0.0"
       },
       "0xd35fC069dcBD50780c706750A41AfC418f25f5D6": {
         "Name": "WhitelistRecoveryFacet",
         "Version": "1.0.2"
+      },
+      "0x9C820C1569ffE99DCF54BBDf3C54682E33a40379": {
+        "Name": "AcrossV4SwapFacet",
+        "Version": ""
```
Comment (mirooon) [inline deployments/arbitrum.diamond.staging.json]: we are still loosing version here. Not only for staging It's present on db so it has to be something with the script diamondUpdateLog. I've just created a ticket for it: 

Db state:
image

File: src/Facets/AcrossV4SwapFacet.sol [hunk 8]
```diff
@@ -0,0 +1,702 @@
...
+        ISpokePoolPeriphery _spokePoolPeriphery,
+        address _spokePool,
+        address _sponsoredOftSrcPeriphery,
+        address _sponsoredCctpSrcPeriphery,
+        address _backendSigner
+    ) {
+        // _sponsoredOftSrcPeriphery and _sponsoredCctpSrcPeriphery can be address(0) as
+        // they are not available on all chains
+        if (
+            address(_spokePoolPeriphery) == address(0) ||
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: constructor does not allow _spokePoolPeriphery to be zero address but in acrossV4Swap.json we have zero addresses for `base` and `boba` networks

File: config/acrossV4Swap.json [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+  "lisk": {
+    "spokePoolPeriphery": "0x89415a82d909a7238d69094C3Dd1dCC1aCbDa85C",
+    "spokePool": "0x9552a0a6624A23B848060AE5901659CDDa1f83f8",
+    "sponsoredOftSrcPeriphery": "0x0000000000000000000000000000000000000000",
+    "sponsoredCctpSrcPeriphery": "0x0000000000000000000000000000000000000000"
+  },
+  "megaeth": {
+    "spokePoolPeriphery": "0xf0aBCe137a493185c5E768F275E7E931109f8981",
+    "spokePool": "0x3db06da8f0a24a525f314eec954fc5c6a973d40e",
+    "sponsoredOftSrcPeriphery": "0x0000000000000000000000000000000000000000",
```
Comment (mirooon) [inline config/acrossV4Swap.json]: oft contract exist: https://github.com/across-protocol/contracts/blob/master/broadcast/deployed-addresses.json#L637-L641

File: src/Facets/AcrossV4SwapFacet.sol [hunk 10]
```diff
@@ -0,0 +1,702 @@
...
+        // Validate amount matches bridgeData
+        _validateAmount(quote.signedParams.amountLD, _bridgeData.minAmount);
+
+        LibAsset.maxApproveERC20(
+            IERC20(_bridgeData.sendingAssetId),
+            SPONSORED_OFT_SRC_PERIPHERY,
+            _bridgeData.minAmount
+        );
+
+        ISponsoredOFTSrcPeriphery(SPONSORED_OFT_SRC_PERIPHERY).deposit{
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:556]: there is no check against if SPONSORED_OFT_SRC_PERIPHERY is zero but can be

File: src/Facets/AcrossV4SwapFacet.sol [hunk 11]
```diff
@@ -0,0 +1,702 @@
...
+            revert InformationMismatch();
+        }
+
+        LibAsset.maxApproveERC20(
+            IERC20(_bridgeData.sendingAssetId),
+            SPONSORED_CCTP_SRC_PERIPHERY,
+            _bridgeData.minAmount
+        );
+
+        ISponsoredCCTPSrcPeriphery(SPONSORED_CCTP_SRC_PERIPHERY)
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:632]: there is no check against if SPONSORED_CCTP_SRC_PERIPHERY is zero but can be

File: src/Facets/AcrossV4SwapFacet.sol [hunk 12]
```diff
@@ -0,0 +1,702 @@
...
+        if (_chainId == 10) return 2; // OP Mainnet
+        if (_chainId == 42161) return 3; // Arbitrum
+        if (_chainId == LIFI_CHAIN_ID_SOLANA) return 5; // Solana
+        if (_chainId == 8453) return 6; // Base
+        if (_chainId == 137) return 7; // Polygon PoS
+        if (_chainId == 130) return 10; // Unichain
+        if (_chainId == 59144) return 11; // Linea
+        if (_chainId == 81224) return 12; // Codex
+        if (_chainId == 146) return 13; // Sonic
+        if (_chainId == 480) return 14; // World Chain
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:751]: what about 15? Monad?

File: src/Facets/AcrossV4SwapFacet.sol [hunk 13]
```diff
@@ -0,0 +1,702 @@
...
+
+            // register max approval in case current approval is insufficient
+            LibAsset.maxApproveERC20(
+                IERC20(_bridgeData.sendingAssetId),
+                address(SPOKE_POOL_PERIPHERY),
+                _bridgeData.minAmount
+            );
+        }
+
+        SPOKE_POOL_PERIPHERY.swapAndBridge{ value: msgValue }(
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:477]: spoke pool periphery contract complitely ignore submissionFees.amount so I think we can do revert here if submissionFees.amount > 0 just in case backend would generate non zero amount

File: src/Facets/AcrossV4SwapFacet.sol [hunk 14]
```diff
@@ -0,0 +1,702 @@
...
+            uint256 refundAmount = _bridgeData.minAmount - _preSwapAmount;
+            // Sponsored CCTP quote does not include a refund recipient; refund to the caller.
+            LibAsset.transferERC20(
+                _bridgeData.sendingAssetId,
+                msg.sender,
+                refundAmount
+            );
+            _bridgeData.minAmount = _preSwapAmount;
+        }
+
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:590]: I think above we could also check sourceDomain from quote agains _chainIdToCctpDomainId(block.chainId)

File: src/Facets/AcrossV4SwapFacet.sol [hunk 15]
```diff
@@ -0,0 +1,702 @@
...
+        _validateDestinationChainId(
+            _bridgeData.destinationChainId,
+            params.destinationChainId
+        );
+
+        // NOTE: Do only use this facet if you trust the Across Swap API.
+        // Across uses a receiver contract on dst chain to execute destination calldata.
+        // We cannot validate the final receiver address here since its somewhere encoded in DepositParams.message.
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:345]: what if we take the raw `message` and try to extract the receiver directly from message by reading the receiver field at a known offset, like it’s done in `_parseReceiver` in `MayanFacet` contract? I think "receiver" will be always in the same place in message

File: src/Facets/AcrossV4SwapFacet.sol [hunk 16]
```diff
@@ -0,0 +1,702 @@
...
+        // Validate depositor/refund address to prevent lost funds
+        if (params.depositor == bytes32(0)) {
+            revert InvalidCallData();
+        }
+
+        // Validate amount matches bridgeData
+        _validateAmount(params.inputAmount, _bridgeData.minAmount);
+
+        // Validate asset matches for ERC20 flows
+        if (!LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: repeated isNativeAsset() checks. Here and in line 336.
I think we can store this value in the variable to save gas

File: src/Facets/AcrossV4SwapFacet.sol [hunk 17]
```diff
@@ -0,0 +1,702 @@
...
+        // If this is the positive-slippage path, refund any surplus and bridge the originally quoted amount.
+        // We MUST NOT change the signed quote amount, otherwise the signature would become invalid.
+        if (_preSwapAmount != 0) {
+            uint256 refundAmount = _bridgeData.minAmount - _preSwapAmount;
+            // Sponsored CCTP quote does not include a refund recipient; refund to the caller.
+            LibAsset.transferERC20(
+                _bridgeData.sendingAssetId,
+                msg.sender,
+                refundAmount
+            );
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: can we accept that we have two behaviours of refunding? With OFT we refund to refundRecipient and with CCTP it defaults to caller?

File: src/Facets/AcrossV4SwapFacet.sol [hunk 18]
```diff
@@ -0,0 +1,702 @@
...
+            _params.fillDeadline,
+            _params.exclusivityParameter,
+            _params.message
+        );
+    }
+
+    /// @notice Converts LiFi internal (non-EVM) chain IDs to Across chain IDs
+    ///         For EVM chainIds there is no need to convert, they will just returned as-is
+    /// @param _destinationChainId The LiFi chain ID to convert
+    function _getAcrossChainId(
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: super nitpick. as it converts to across chain id imo `_toAcrossChainId` naming for this function will be more clear

File: src/Facets/AcrossV4SwapFacet.sol [hunk 19]
```diff
@@ -0,0 +1,702 @@
...
+        if (_chainId == 59144) return 11; // Linea
+        if (_chainId == 81224) return 12; // Codex
+        if (_chainId == 146) return 13; // Sonic
+        if (_chainId == 480) return 14; // World Chain
+        if (_chainId == 1329) return 16; // Sei
+        if (_chainId == 50) return 18; // XDC
+        if (_chainId == 999) return 19; // HyperEVM
+        if (_chainId == 1337) return 19; // HyperCore (via HyperEVM)
+        if (_chainId == 57073) return 21; // Ink
+        if (_chainId == 98866) return 22; // Plume
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:758]: worth to double check if we can get ids for megaeth and plasma. In config they are covered

File: src/Facets/AcrossV4SwapFacet.sol [hunk 20]
```diff
@@ -0,0 +1,702 @@
...
+        } else if (swapApiTarget == SwapApiTarget.SponsoredOFTSrcPeriphery) {
+            _callSponsoredOftDeposit(_bridgeData, callData, _preSwapAmount);
+        } else if (swapApiTarget == SwapApiTarget.SponsoredCCTPSrcPeriphery) {
+            _callSponsoredCctpDepositForBurn(
+                _bridgeData,
+                callData,
+                _preSwapAmount
+            );
+        } else {
+            revert InvalidCallData();
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: line not covered in the tests

File: src/Facets/AcrossV4SwapFacet.sol [hunk 21]
```diff
@@ -0,0 +1,702 @@
...
+            abi.encodePacked("\x19\x01", _domainSeparator(), structHash)
+        );
+
+        address recoveredSigner = ECDSA.recover(
+            digest,
+            _acrossV4SwapFacetData.signature
+        );
+
+        if (recoveredSigner != BACKEND_SIGNER) {
+            revert InvalidSignature();
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol:309]: line not covered in the tests

File: src/Facets/AcrossV4SwapFacet.sol [hunk 22]
```diff
@@ -0,0 +1,702 @@
...
+
+        // depositForBurn is not payable
+        if (msg.value != 0) revert InvalidCallData();
+
+        // Validate amount matches bridgeData
+        _validateAmount(quote.amount, _bridgeData.minAmount);
+
+        // Validate burnToken matches bridgeData asset
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            revert InvalidCallData();
```
Comment (mirooon) [inline src/Facets/AcrossV4SwapFacet.sol]: line not covered in the tests

File: src/Facets/AcrossV4SwapFacet.sol [hunk 23]
```diff
@@ -0,0 +1,702 @@
...
+        }
+
+        if (_recipient == bytes32(0)) revert InvalidReceiver();
+        if (
+            _enforceEvmMatch &&
+            _recipient != _convertAddressToBytes32(_bridgeData.receiver)
+        ) {
+            revert InvalidReceiver();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:700]: we would have to maintain a list of non-EVM chainIds for this guard to work. That doesnt make any sense. Any other ideas?

File: deployments/arbitrum.diamond.staging.json (hunk 7 above)
Comment (0xDEnYO) [reply deployments/arbitrum.diamond.staging.json]: I found out why. If you deploy something new and then update the diamond log, the script pulls from the local cache file and that is outdated (TTL=5min). 
I will open a PR that resolves this by updating the cache after each mongodb write.
I think it's OK here cause it will be updated very soon by the next PR.

File: config/acrossV4Swap.json (hunk 9 above)
Comment (0xDEnYO) [reply config/acrossV4Swap.json]: added

File: src/Facets/AcrossV4SwapFacet.sol (hunk 10 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:556]: then the tx would revert. 
We trust our BE that it will only produce calldata that makes sense. The signature ensures that only BE-generated calldata can be used.
I feel OK without it. WDYT?

File: src/Facets/AcrossV4SwapFacet.sol (hunk 11 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:632]: same as above

File: src/Facets/AcrossV4SwapFacet.sol (hunk 12 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:751]: good catch, added

File: src/Facets/AcrossV4SwapFacet.sol (hunk 15 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:345]: No, the receiver is not always in the same spot as the calldata differs depending on what swaps are done etc.

File: src/Facets/AcrossV4SwapFacet.sol (hunk 18 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: renamed

File: src/Facets/AcrossV4SwapFacet.sol (hunk 19 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:758]: The link of CCTP domains is in the function description: 
https://developers.circle.com/cctp/concepts/supported-chains-and-domains#cctp-v2-supported-domains

no support for megaeth or plasma there

File: script/demoScripts/demoAcrossV4Swap.ts [hunk 24]
```diff
@@ -0,0 +1,780 @@
...
+    bridgeData.minAmount,
+    publicClient
+  )
+  await ensureAllowance(
+    tokenContract,
+    walletAddress,
+    lifiDiamondAddress as string,
+    bridgeData.minAmount,
+    publicClient
+  )
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/demoScripts/demoAcrossV4Swap.ts]: fixed, pls check

File: src/Facets/AcrossV4SwapFacet.sol [hunk 25]
```diff
@@ -0,0 +1,702 @@
...
+
+        LibAsset.maxApproveERC20(
+            IERC20(_bridgeData.sendingAssetId),
+            SPONSORED_OFT_SRC_PERIPHERY,
+            _bridgeData.minAmount
+        );
+
+        ISponsoredOFTSrcPeriphery(SPONSORED_OFT_SRC_PERIPHERY).deposit{
+            value: msg.value
+        }(quote, signature);
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:558]: not an issue in this case

File: src/Facets/AcrossV4SwapFacet.sol (hunk 20 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: initially I wasnt able to do it (cause Solidity compiler doesnt allow to send invalid values) but then I thought about abi.encoding a call with an invalid value and that worked.
Added.

File: src/Facets/AcrossV4SwapFacet.sol (hunk 8 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: updated

File: src/Facets/AcrossV4SwapFacet.sol (hunk 13 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:477]: Feels like a "nice to have" validation for me which we usually dont do (just what we need to keep funds safe). 

Or am I missing or misunderstanding sth? @mirooon

File: src/Facets/AcrossV4SwapFacet.sol (hunk 14 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:590]: We could, but do we have to? If yes, why?
This path is restricted to calldata from our backend via BE signature.

File: src/Facets/AcrossV4SwapFacet.sol (hunk 16 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: good catch, updated

File: src/Facets/AcrossV4SwapFacet.sol [hunk 26]
```diff
@@ -0,0 +1,699 @@
...
+            _callSponsoredOftDeposit(_bridgeData, callData, _preSwapAmount);
+        } else if (swapApiTarget == SwapApiTarget.SponsoredCCTPSrcPeriphery) {
+            _callSponsoredCctpDepositForBurn(
+                _bridgeData,
+                callData,
+                _preSwapAmount
+            );
+        } else {
+            revert InvalidCallData();
+        }
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:262]: no need to, silent reverts are OK

File: src/Facets/AcrossV4SwapFacet.sol (hunk 22 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: was not testable...which led me to removing it as it's a case that is already caught before by other validations

File: src/Facets/AcrossV4SwapFacet.sol (hunk 21 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:309]: covered now

File: src/Facets/AcrossV4SwapFacet.sol (hunk 17 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol]: good point, I added a refund recipient parameter, pls review

File: src/Facets/AcrossV4SwapFacet.sol (hunk 10 above)
Comment (mirooon) [reply src/Facets/AcrossV4SwapFacet.sol:556]: actually this value is set on deployment and we allow it to be set to zero and yes the tx would revert if its zero address I am good not having a check

File: src/Facets/AcrossV4SwapFacet.sol (hunk 14 above)
Comment (mirooon) [reply src/Facets/AcrossV4SwapFacet.sol:590]: in `_verifySignatureIfRequired` we do not check the backend signature for cctp if the quote is cctp the function returns early and skips the signature check this is risky because the contract does not care which chain the quote is submitted on this could lead to accidental loss so i think we should add the signature check for cctp

File: src/Facets/AcrossV4SwapFacet.sol (hunk 13 above)
Comment (mirooon) [reply src/Facets/AcrossV4SwapFacet.sol:477]: across contract function `swapAndBridge` completely ignores the `submissionFees` struct. It does not look at it at all. In case if submissionFees.amount won't be zero for this path user's funds can be stuck but we rely on calldata from the backend so I agree it's nice to have

File: script/helperFunctions.sh [hunk 27]
```diff
@@ -999,23 +1031,54 @@ function saveDiamondPeriphery() {
...
+      warning "[$NETWORK] Merge failed, creating fresh diamond structure with periphery"
+      result=$(jq -n --arg diamond_name "$DIAMOND_NAME" --argjson periphery_obj "$PERIPHERY_JSON" '
+        {
+          ($diamond_name): {
+            Facets: {},
+            Periphery: $periphery_obj
+          }
+        }
+      ' 2>/dev/null)
+    fi
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/helperFunctions.sh]: outdated? pls confirm

File: src/Facets/AcrossV4SwapFacet.sol (hunk 13 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:477]: ok, in this case I would leave it as-is

File: src/Facets/AcrossV4SwapFacet.sol (hunk 10 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:556]: ok

File: test/solidity/Facets/Across/SwapAPI/AcrossV4SwapFacet.Sponsored.t.sol [hunk 28]
```diff
@@ -0,0 +1,598 @@
...
+            localBridgeData,
+            AcrossV4SwapFacet.AcrossV4SwapFacetData({
+                swapApiTarget: AcrossV4SwapFacet
+                    .SwapApiTarget
+                    .SponsoredOFTSrcPeriphery,
+                callData: abi.encode(quote, signature),
+                signature: ""
+            })
+        );
+        vm.stopPrank();
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/Facets/Across/SwapAPI/AcrossV4SwapFacet.Sponsored.t.sol:372]: added, pls check

[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/Facets/Across/SwapAPI/AcrossV4SwapFacet.Sponsored.t.sol:372]: check again now

File: src/Facets/AcrossV4SwapFacet.sol (hunk 14 above)
Comment (0xDEnYO) [reply src/Facets/AcrossV4SwapFacet.sol:590]: yes sorry, my bad. I confused the paths. It's not signature-protected. 
However, I do not really see how loss of funds can occur. 

The SponsoredCCTPSrcPeriphery validates that source domain is correct, we dont need to do it again. 

image

OK for you?

Comment (mirooon) [reply src/Facets/AcrossV4SwapFacet.sol:590]: sure ok!

**Review by mirooon — DISMISSED** [review]
